include/openmc/position.h

```diff
@@ -63,6 +63,13 @@ struct Position {
     return std::sqrt(x*x + y*y + z*z);
   }
 
+  inline Position reflect(Position n) {
+    const double projection = n.dot(*this);
+    const double magnitude = n.dot(n);
+    n *= (2.0 * projection / magnitude);
+    return *this -= n;
```

Comment:
```suggestion
    return *this - n;
```
To me, it's a little weird that this method both modifies the direction and returns the new direction. I'd prefer to see this just be a `const` method that returns the direction; if the user wants to change the direction, they can do
```C++
u = u.reflect(n)
```